R/format_data.R

```diff
@@ -1702,29 +1809,42 @@ fmt_datetime <- function(data,
                          columns,
                          rows = everything(),
                          date_style = 2,
-                         time_style = 2) {
+                         time_style = 2,
+                         sep = " ",
+                         format = NULL,
+                         tz = "UTC",
+                         pattern = "{x}") {
 
   # Perform input object validation
   stop_if_not_gt(data = data)
 
-  # Transform `date_style` to `date_format`
-  date_format_str <- get_date_format(date_style = date_style)
+  if (!is.null(format)) {
 
-  # Transform `time_style` to `time_format`
-  time_format_str <- get_time_format(time_style = time_style)
+    # Ensure that the format string meets some basic validation requirements
+    check_format_string(format = format)
+
+  } else {
+
+    # Transform `date_style` to `date_format`
+    date_format_str <- get_date_format(date_style = date_style)
+
+    # Transform `time_style` to `time_format`
+    time_format_str <- get_time_format(time_style = time_style)
+  }
 
   # Stop function if any columns have data that is incompatible
   # with this formatter
   if (
     !column_classes_are_valid(
       data = data,
       columns = {{ columns }},
-      valid_classes = "character"
-    )) {
-    stop(
-      "The `fmt_datetime()` function can only be used on `columns` with `character` values.",
-      call. = FALSE
-    )
+      valid_classes = c("Date", "POSIXt", "character"))
```

Comment:
If we don't accept `POSIXlt` as input, then valid class should be `"POSIXct"`

Reply:
Now changed.